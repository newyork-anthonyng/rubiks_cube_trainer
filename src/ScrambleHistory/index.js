import React, { Fragment } from "react";
import PropTypes from "prop-types";

const ScrambleHistory = ({ history }) => (
  <Fragment>
    <h2>Scramble History</h2>
    {history.length === 0 ? (
      <p>No scramble history available</p>
    ) : (
      <ul>
        {history.map(({ scramble, difficulty }) => (
          <li key={scramble}>
            <span>{difficulty}: </span>
            <span>{scramble}</span>
          </li>
        ))}
      </ul>
    )}
  </Fragment>
);

ScrambleHistory.defaultProps = {
  history: []
};

ScrambleHistory.propTypes = {
  history: PropTypes.arrayOf(
    PropTypes.shape({
      scramble: PropTypes.string,
      difficulty: PropTypes.number
    })
  )
};

export default ScrambleHistory;
