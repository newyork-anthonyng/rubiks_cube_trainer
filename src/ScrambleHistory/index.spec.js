import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import ScrambleHistory from "./";

const defaultProps = {
  history: [{ difficulty: 4, scramble: "LOL" }]
};

it("should render correctly when history contains information", () => {
  const wrapper = shallow(<ScrambleHistory {...defaultProps} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render correctly if history is an empty list", () => {
  const wrapper = shallow(<ScrambleHistory {...defaultProps} history={[]} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should not explode when history is undefined", () => {
  const wrapper = shallow(
    <ScrambleHistory {...defaultProps} history={undefined} />
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
