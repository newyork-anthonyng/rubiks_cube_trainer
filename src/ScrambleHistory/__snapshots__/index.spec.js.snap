// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not explode when history is undefined 1`] = `
<React.Fragment>
  <h2>
    Scramble History
  </h2>
  <p>
    No scramble history available
  </p>
</React.Fragment>
`;

exports[`should render correctly if history is an empty list 1`] = `
<React.Fragment>
  <h2>
    Scramble History
  </h2>
  <p>
    No scramble history available
  </p>
</React.Fragment>
`;

exports[`should render correctly when history contains information 1`] = `
<React.Fragment>
  <h2>
    Scramble History
  </h2>
  <ul>
    <li
      key="LOL"
    >
      <span>
        4
        : 
      </span>
      <span>
        LOL
      </span>
    </li>
  </ul>
</React.Fragment>
`;
